SimG4CMS/PPS/interface/TotemRPSD.h

```diff
@@ -0,0 +1,125 @@
+/** fake sensitive detector modelled after PSimHit for a user example;
+    the ORCA/CommonDet/BasicDet/interface/PSimHit.h and
+    ORCA/CommonDet/PBasicDet/interface/PSimHitROUFactory.h
+    are copied under Mantis/USDHitExample/test/stubs;
+    the user must provide links to ORCA libraries PBasicDet, BasicDet and
+    DetGeometry which must be loaded before the USDHitExample library -
+    see sens.macro under Mantis/G4Notification/test
+ */
+
+#ifndef PPS_TotemRPSD_h
+#define PPS_TotemRPSD_h
+
+/**
+   In this example the TotemSensitiveDetector serves as a master for two different ExampleSD
+ */
+
+#include "SimG4CMS/PPS/interface/TotemRPG4Hit.h"
+#include "SimG4CMS/PPS/interface/TotemRPG4HitCollection.h"
+#include "SimG4CMS/PPS/interface/TotemRPVDetectorOrganization.h"
+
+#include "SimG4Core/Notification/interface/Observer.h"
+#include "SimG4Core/SensitiveDetector/interface/SensitiveTkDetector.h"
+#include "SimG4Core/Notification/interface/BeginOfEvent.h"
+#include "SimG4Core/Notification/interface/EndOfEvent.h"
+
+#include <string>
+
+class G4Step;
+class G4HCofThisEvent;
+class G4Track;
+class G4StepPoint;
+class TrackingSlaveSD;
+class SimTrackManager;
+class TotemTestHitHBNtuple;
+
+class TotemRPSD : public SensitiveTkDetector, public Observer<const BeginOfEvent*>, public Observer<const EndOfEvent*> {
+public:
+  TotemRPSD(const std::string&,
+            const edm::EventSetup&,
+            const SensitiveDetectorCatalog&,
+            edm::ParameterSet const&,
+            const SimTrackManager*);
+  ~TotemRPSD() override;
+
+  void Print_Hit_Info();
+
+  void Initialize(G4HCofThisEvent* HCE) override;
+  void EndOfEvent(G4HCofThisEvent* eventHC) override;
+  void clear() override;
+  void DrawAll() override;
+  void PrintAll() override;
+
+  void fillHits(edm::PSimHitContainer&, const std::string&) override;
+  static constexpr double rp_garage_position_ = 40.0;
+
+private:
+  void clearHits() override;
+  bool ProcessHits(G4Step* step, G4TouchableHistory* tHistory) override;
+  uint32_t setDetUnitId(const G4Step* step) override;
+  void update(const BeginOfEvent*) override;
+  void update(const ::EndOfEvent*) override;
+
+  void SetNumberingScheme(TotemRPVDetectorOrganization* scheme);
+
+  TrackingSlaveSD* slave;
+  TotemRPVDetectorOrganization* numberingScheme;
+
+private:
+  int verbosity_;
+
+  G4ThreeVector SetToLocal(G4ThreeVector globalPoint);
+  void GetStepInfo(G4Step* aStep);
```

Comment:
@mundim , better to use only const pointers to G4Step - I do not think you plan to kill any of particles in PPS.

Reply:
Done.